### PR TITLE
feat(ddl): support `create_schema` and implement `create_database` as an abstract base class

### LINKED ISSUE #6331: feat(ddl): fill out support for databases and schemas
### Is your feature request related to a problem?

I'd like to have a consistent interface for exploring and accessing the hierarchy of a given backend's databases.

Here's a table showing each backend's support for two levels of hierarchy: `database -> schema` and if the name differs in the backend from one or more of those levels I list it:

|    **backend** | **database** | **schema** | **distinct from user** |
|---------------:|--------------|------------| ---------------------- |
|   **bigquery** | project      | dataset    | Y                      |
| **clickhouse** | database     |            |                        |
|       **dask** | NA           | NA         |                        |
| **datafusion** | database     | schema     | Y                      |
|      **druid** | connector    |            |                        |
|     **duckdb** | database     | schema     | Y                      |
|     **impala** | database     |            |                        |
|      **mssql** | database     | schema     | Y                      |
|      **mysql** | database     |            |                        |
|     **oracle** | database     | user (?)   | N                      |
|     **pandas** | NA           |            |                        |
|     **polars** | NA           |            |                        |
|   **postgres** | database     | schema     | Y                      |
|    **pyspark** | database     |            |                        |
|  **snowflake** | database     | schema     | Y                      |
|     **sqlite** | database     |            |                        |
|      **trino** | catalog      | schema     | Y                      |

Looking at this table it appears that most databases that have support for a second level of hierarchy that is, or can reasonably be called `schema`.

### Describe the solution you'd like

I'd like to be able to do the follow where it makes sense for the backend:

1. Create databases
2. Create schemas
3. List databases
4. List schemas

Given the above table, I think it makes sense to make existing `create_database()`/`drop_database()`/`list_databases()` consistent with that table, and add a `create_schema()`/`drop_schema()` for the backends where that makes sense.

### What version of ibis are you running?

`master`

### What backend(s) are you using, if any?

NA

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Adds support for databases and schema creation and listing.

Closes #6331.

### CODE REVIEW

File: ibis/backends/bigquery/__init__.py [hunk 1]
```diff
@@ -231,6 +231,45 @@ def project_id(self):
...
+        self,
+        name: str,
+        database: str | None = None,
+        force: bool = False,
+        collate: str | None = None,
+        **options: Any,
+    ) -> None:
+        code = "CREATE SCHEMA"
+        if force:
+            code += " IF NOT EXISTS"
```
Comment (gforsyth) [inline ibis/backends/bigquery/__init__.py:244]: the wording is hard to parse for me, but https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#arguments says

```
IF NOT EXISTS: If any dataset exists with the same name, the CREATE statement has no effect.
```

Should we interpret `force` here to mean "don't error" or "overwrite" (or something else)?

Comment (gforsyth) [reply ibis/backends/bigquery/__init__.py:244]: looking at `drop_schema` I think it's "don't error" -- and I'm having deja vu that we've discussed this at some point in the past.  

I think this is probably how `exists_ok` became a keyword argument in pathlib, but I don't know that we want to copy that.

File: ibis/backends/bigquery/__init__.py [hunk 2]
```diff
@@ -231,6 +231,45 @@ def project_id(self):
...
 
+    def create_schema(
+        self,
+        name: str,
+        database: str | None = None,
+        force: bool = False,
+        collate: str | None = None,
+        **options: Any,
+    ) -> None:
+        code = "CREATE SCHEMA"
```
Comment (gforsyth) [inline ibis/backends/bigquery/__init__.py:242]: Not a hill anyone has to die on, but what do you think about calling this (and the related) variables `create_stmt` and `drop_stmt`, respectively?

**Review by gforsyth — COMMENTED** [review]
implementation looks good and I'm all for making these consistent and available.

not sure what to do about `force` as a keyword argument.

File: ibis/backends/bigquery/__init__.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/bigquery/__init__.py:242]: Done!

File: ibis/backends/bigquery/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/bigquery/__init__.py:244]: I'd rather keep it consistent with the current interfaces, which all use `force` even if `force` is suboptimal.

**Review by gforsyth — COMMENTED** [review]
Implementation looks good!  I flagged a few docstring things that need fixing (I think).

I don't like `force`, but I think we can change that in a follow-up PR (and possibly following a deprecation cycle).

File: ibis/backends/base/__init__.py [hunk 3]
```diff
@@ -519,6 +519,72 @@ def to_delta(
...
+        """
+
+
+class CanCreateSchema(abc.ABC):
+    @abc.abstractmethod
+    def create_schema(
+        self, name: str, database: str | None = None, force: bool = False
+    ) -> None:
+        """Create a schema named `name` in `database`."""
+
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/base/__init__.py]: Done!

File: ibis/backends/base/__init__.py [hunk 4]
```diff
@@ -519,6 +519,72 @@ def to_delta(
...
+    @abc.abstractmethod
+    def drop_schema(
+        self, name: str, database: str | None = None, force: bool = False
+    ) -> None:
+        """Drop the schema with `name` in `database`."""
+
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/base/__init__.py]: Done!

**Review by gforsyth — APPROVED** [review]
:shipit:
